library/src/androidTest/java/OperationUIDataTests.kt

```diff
@@ -0,0 +1,456 @@
+            return
+        }
+    }
+
+
+    @Test
+    fun testPostApprovalResponseRedirect() {
+        val result = prepareResult(postApprovalResponseRedirect)
+            ?: run {
+                fail("Failed to parse JSON data")
+                return
+            }
+
+        val ui = OperationUIData(
+            flipButtons = null,
+            blockApprovalOnCall = null,
+            preApprovalScreen = null,
+            postApprovalScreen = PostApprovalScreenRedirect(
+                heading = "Thank you for your order",
+                message = "You will be redirected to the merchant application.",
+                payload = RedirectPostApprovalScreenPayload(
+                    text = "Go to the application",
+                    url = "https://www.alza.cz/ubiquiti-unifi-ap-6-pro-d7212937.htm",
+                    countdown = 5
+                )
+            )
+        )
+
+        assertEquals(result.ui?.flipButtons, ui.flipButtons)
+        assertEquals(result.ui?.blockApprovalOnCall, ui.blockApprovalOnCall)
+        assertEquals(result.ui?.preApprovalScreen?.type, ui.flipButtons)
+        assertEquals((result.ui?.postApprovalScreen as? PostApprovalScreenReview)?.heading, (ui.postApprovalScreen as? PostApprovalScreenReview)?.heading)
+        assertEquals((result.ui?.postApprovalScreen as? PostApprovalScreenReview)?.message, (ui.postApprovalScreen as? PostApprovalScreenReview)?.message)
+        assertEquals((result.ui?.postApprovalScreen as? PostApprovalScreenReview)?.payload?.attributes, (ui.postApprovalScreen as? PostApprovalScreenReview)?.payload?.attributes)
+    }
+
+    @Test
+    fun testPostApprovalResponseReview() {
+        val result = prepareResult(postApprovalResponseReview)
+            ?: run {
+                fail("Failed to parse JSON data")
+                return
+            }
+
+        val ui = OperationUIData(
+            flipButtons = null,
+            blockApprovalOnCall = null,
+            preApprovalScreen = null,
+            postApprovalScreen = PostApprovalScreenReview(
+                heading = "Successful",
+                message = "The operation was approved.",
+                payload = ReviewPostApprovalScreenPayload(
+                    listOf<Attribute>(
+                        NoteAttribute(
+                            note = "myNote",
+                            label = Attribute.Label(
+                                id = "1",
+                                value = "test label"
+                            )
+                        )
+                    )
+                )
+            )
+        )
+
+        assertEquals(result.ui?.flipButtons, ui.flipButtons)
+        assertEquals(result.ui?.blockApprovalOnCall, ui.blockApprovalOnCall)
+        assertEquals(result.ui?.preApprovalScreen?.type, ui.flipButtons)
+        assertEquals((result.ui?.postApprovalScreen as? PostApprovalScreenReview)?.heading, (ui.postApprovalScreen as? PostApprovalScreenReview)?.heading)
```

Comment:
You're calling `result.ui?.postApprovalScreen as? PostApprovalScreenReview)` 5 times here, it would be much cleaner to make a variable out of it

Reply:
👍